services/CourseManager.php

```diff
@@ -148,8 +165,7 @@ public function setModuleCanBeOpenedByLearner(Learner $learner, Course $course,
                     || $this->learnerManager->hasBeenOpenedBy($course, $previousModule, $previousActivity, $learner)
```

Comment:
si le mode scénarisation des modules est à ON, tu regardes là seulement si la dernière activité du précédent module a été accédé mais il faut plutôt regarder que le module précédent a été fini.

On n'a par contre pas directement si un utilisateur a terminé un module avec Progresses. Qd j'avais fais ProgressDashboard, j'avais essayé de le faire mais vu que ça faisait bcp de traitements sur tous les utilisateurs, j'avais fais le choix de le faire plus optimisé mais non défini au niveau de Progresses. 
On peut utiliser par contre getUsernamesForFinishedActivity qui lancé sur les progresses de l'apprenant rend un tableau avec un élément si l'utilisateur a finis l'activité. Il faudrait alors itérer sur les activité d'un module et rendre false dès la première activité non terminée, et vrai à la fin de la méthode.
Cette méthode a à mon avis plus sa place dans LearnerManager et pourrait se nommer hasModuleBeenFinished.

Reply:
>mais il faut plutôt regarder que le module précédent a été fini.

ça me va comme spec. à implémenter effectivement.

Reply:
J'ai fait un correctif pour tester si le module précédent est terminé https://github.com/YesWiki/yeswiki-extension-lms/commit/b40abafb9ec9cf6cc53e1eb41defdd080213a4cb
mais je te laisse prendre la main dès que je te fais signe pour adapter à ta guise.